### PR TITLE
Dartdoc new

### PR DESCRIPTION
**Pull request by munificent**

### CODE REVIEW

**Comment by chalin** [comment]
I believe that this was built atop #819. Maybe we should get #819 merged first. /cc @kwalrath

**Comment by munificent** [comment]
Oh, crap, sorry about that. I meant to switch back to master first but must have forgotten.

File: src/_guides/language/effective-dart/documentation.md [hunk 1]
```diff
@@ -277,21 +277,32 @@ makes an API easier to learn.
...
 {:.good-style}
-<?code-excerpt "misc/lib/effective_dart/docs_good.dart (new)"?>
+<?code-excerpt "misc/lib/effective_dart/docs_good.dart (member)"?>
 {% prettify none %}
-/// To create a point, call [new Point] or use [new Point.polar] to ...
+/// Similar to [Duration.inDays], but handles fractional days.
+{% endprettify %}
+
+Named constructors work like members. For the unnamed constructor, put
+parentheses after the class name:
```
[the remark this answers is not in the document]
Comment (munificent) [reply src/_guides/language/effective-dart/documentation.md]: "as you would" sounds a little stiff to me, but I tried a different wording.

[the remark this answers is not in the document]
Comment (chalin) [reply src/_guides/language/effective-dart/documentation.md]: Agreed, but I couldn't think of anything else. I like your new sentence!

File: src/_guides/language/effective-dart/documentation.md [hunk 2]
```diff
@@ -277,21 +277,32 @@ makes an API easier to learn.
...
 <?code-excerpt "misc/lib/effective_dart/docs_good.dart (identifiers)"?>
 {% prettify none %}
 /// Throws a [StateError] if ...
-/// similar to [anotherMethod], but ...
+/// similar to [anotherMethod()], but ...
 {% endprettify %}
 
-You can also link to constructors using `new` followed by the constructor:
+You can put parentheses after the name too, which makes it clearer when you're
+referring to a method or constructor. You can link to a specific member of a
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/documentation.md]: This last sentence should be its own paragraph. Maybe:

To link to a specific member of a class (from another class, for example), use the class name...

File: src/_guides/language/effective-dart/documentation.md [hunk 3]
```diff
@@ -277,21 +277,32 @@ makes an API easier to learn.
...
 
 {:.good-style}
-<?code-excerpt "misc/lib/effective_dart/docs_good.dart (new)"?>
+<?code-excerpt "misc/lib/effective_dart/docs_good.dart (member)"?>
 {% prettify none %}
-/// To create a point, call [new Point] or use [new Point.polar] to ...
+/// Similar to [Duration.inDays], but handles fractional days.
+{% endprettify %}
+
+Named constructors work like members. For the unnamed constructor, put
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/documentation.md]: The first sentence seems too imprecise. How about:

Refer to named constructors the same way you refer to methods. To refer to the unnamed constructor, ...

File: src/_guides/language/effective-dart/documentation.md [hunk 4]
```diff
@@ -277,21 +277,32 @@ makes an API easier to learn.
 ### DO use square brackets in doc comments to refer to in-scope identifiers.
 
 If you surround things like variable, method, or type names in square brackets,
-then dartdoc will look up the name and link to its docs.
+then dartdoc looks up the name and links to the entity's docs:
```
[the remark this answers is not in the document]
Comment (munificent) [reply src/_guides/language/effective-dart/documentation.md]: Done.

File: src/_guides/language/effective-dart/documentation.md [hunk 5]
```diff
@@ -277,21 +277,32 @@ makes an API easier to learn.
...
 {:.good-style}
 <?code-excerpt "misc/lib/effective_dart/docs_good.dart (identifiers)"?>
 {% prettify none %}
 /// Throws a [StateError] if ...
-/// similar to [anotherMethod], but ...
+/// similar to [anotherMethod()], but ...
 {% endprettify %}
 
-You can also link to constructors using `new` followed by the constructor:
+You can put parentheses after the name too, which makes it clearer when you're
```
[the remark this answers is not in the document]
Comment (munificent) [reply src/_guides/language/effective-dart/documentation.md]: Reworded and moved up to the previous paragraph.

File: src/_guides/language/effective-dart/documentation.md (hunk 2 above)
Comment (munificent) [reply src/_guides/language/effective-dart/documentation.md]: Done.

File: src/_guides/language/effective-dart/documentation.md [hunk 6]
```diff
@@ -277,7 +277,9 @@ makes an API easier to learn.
 ### DO use square brackets in doc comments to refer to in-scope identifiers.
 
 If you surround things like variable, method, or type names in square brackets,
-then dartdoc looks up the name and links to the entity's docs:
+then dartdoc looks up the name and links to the relevant API docs. Parentheses
+are optional, but can it clearer when you're referring to a method or
```
Comment (kwalrath) [inline src/_guides/language/effective-dart/documentation.md]: can it -> can make it

😆

**Pull request merged**